### DISCUSSION TITLE
ParalleForeachBody

### DISCUSSION THREAD

**Discussion by mremadevi** [elsa-workflows/elsa-core#5328] [Q&A]
How to pass iterated item as input to ParalleForeachBody

**Comment by glime-ai** [comment]
To pass the iterated item as input to the `ParallelForEach` body in Elsa Workflow, you can use the `CurrentItem` variable that is automatically created by the `ParallelForEach` activity. This variable represents the current item in the loop and is part of the activity execution context for each iteration.

Here's a general approach to using `CurrentItem`:

1. In your `ParallelForEach` activity, specify the collection you want to iterate over.
2. Inside the body of the `ParallelForEach`, you can access the `CurrentItem` variable, which holds the current item of the iteration.
3. Use this `CurrentItem` variable as an input to the activities within the body of the `ParallelForEach`.

For example, if you have an activity that needs the current item as input, you can reference `CurrentItem` directly in that activity's input properties.

Please note that if you have nested `ParallelForEach` activities, each `ParallelForEach` will have its own `CurrentItem` variable in its scope, as mentioned in the GitHub issue #4312.

If you need further clarification or have a specific scenario in mind, please provide more details so I can assist you better.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=dRbvTb7IhCSYiI&utm_medium=github).

**Reply by mremadevi** [reply]
Used Variables.CurrentValue
